[MRG+2] logit -> logistic in plot_logistic.py and minor visual improvements to the plot.

Deborah-Digges opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#Contributing-Pull-Requests
-->
#### Reference Issue

No issue exists for this change currently
#### What does this implement/fix? Explain your changes.

This is a small fix to documentation, correcting the name of a function.
#### Any other comments?

The function `1 / (1 + np.exp(-x))` is the logistic function. 
The logit function is the inverse of the logistic function : `log(x/(1-x))`. I've corrected this in one of the examples.

Deborah-Digges committed:
Small correction of logit->logistic

The function `1 / (1 + np.exp(-x))` is the logistic function. 
The logit function is the inverse of the logistic function : `log(x/(1-x))`

raghavrv commented:
It's indeed a logistic function. Thanks @Deborah-Digges!

Could you also add some labels to the plot (set different colors for the linear regression and logistic regression)? :)

raghavrv commented:
(I'm adding this to 0.18.1 milestone)

Deborah-Digges commented:
Sure @raghavrv !
Would you like me to open a separate PR for it?

raghavrv commented:
You could do that in this PR itself. Add an additional commit and push :)

Deborah-Digges commented:
Sure. 👍 
I'll update this PR soon. Thank you:)

Commit:
Add axes ticks, legend and colors to the plot

Commit:
Fix flake8 errors

Deborah-Digges commented:
<img width="414" alt="screen shot 2016-10-24 at 9 46 14 pm" src="https://cloud.githubusercontent.com/assets/3866405/19653761/51d0df3e-9a33-11e6-9951-55ca258597a5.png">
- Used different colors for logistic and linear plots
- Added legend
- Added ticks to the Axes

raghavrv commented on examples/linear_model/plot_logistic.py:
```diff
@@ -57,9 +57,10 @@ def model(x):
 plt.ylim(-.25, 1.25)
 plt.xlim(-4, 10)
-
+plt.legend(('Logistic Curve', 'Least Squares Line'), loc="lower right",
```
Could you label these as 
* `Logistic Regression model` and
* `Linear Regression model`

raghavrv commented:
Besides the comment, this looks good to me... Thanks!

Commit:
Rename legend labels as per review

Deborah-Digges commented:
@raghavrv done 👍

raghavrv commented:
@agramfort @TomDLT A second look please?

TomDLT commented:
LGTM, waiting for continuous integration then I'll merge

amueller commented:
travis fails pep8.

Commit:
Fix pep8 error

Deborah-Digges commented:
@raghavrv @TomDLT @amueller I've fixed the pep8 errors. 

All checks seem to be passing now 👍

TomDLT commented:
thanks @Deborah-Digges :smiley:
